### ISSUE TITLE
When I give a share link to someone, they should be able to view the file in their browser

### ISSUE THREAD

**Issue by mortie** [nextcloud/server#3300]
I don't know if this is the correct channel for feature requests, but I'll try, and if it isn't, please point me to it. I have ignored the issues submission form, since this isn't a bug report.

When sharing a file or directory containing source code to other people, the only way for them to view the files is to download them and open them in some text editor. It would be really useful to be able to point people to a shared file, and then let them just view it directly in the browser. Because nextcloud supports sharing entire directories, it is almost perfect for letting people see code or documents I'm working on, but the lack of this feature makes that needlessly annoying for the person/people I'm sharing stuff to.

As a quick fix, adding a button which serves the files without the `Content-Disposition: attachment` header would work for most text files, assuming they're served with `Content-Type: text/<whatever>`. A better solution, which would require more time, would be to add a way for apps to provide viewers for shared files.

**TL;DR**: Currently, guests viewing files shared through share link have to download files to see them. An option to view the files in the browser would be useful.

**Comment by nickvergessen** [comment]
Basically the public page is missing all the viewers and editors atm.

**Comment by SuperSayu** [comment]
I agree with a slightly different reason.  I have a friend with a webcomic who was using Dropbox on the backend as an image host, using DB's public folder and public links feature which is on the way out right now.  If a shared Nextcloud folder had similar, easy public file links, I would recommend it as a solution.  As-is, I am not aware of any way to link to images for display on a webpage, using NC as the backend.

An easy raw-access link for publicly visible files (serving media to the browser and downloading other binaries) in addition to the download-specific link would be fantastic.

**Comment by MorrisJobke** [comment]
> An easy raw-access link for publicly visible files (serving media to the browser and downloading other binaries) in addition to the download-specific link would be fantastic.

This is addressed in #2523

The text editor thing is a duplicate of #1571
